pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -120,33 +120,59 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 
 	// If the UserAccount has not been deleted, create or update user and identity resources.
 	// If the UserAccount has been deleted, delete secondary resources identity and user.
+
 	if !util.IsBeingDeleted(userAcc) {
+		reqLogger.Info("Adding finalizer on UserAccount")
 		// Add the finalizer if it is not present
 		if err := r.addFinalizer(userAcc); err != nil {
 			return reconcile.Result{}, err
 		}
-
+	}
+	if !util.IsBeingDeleted(userAcc) && !userAcc.Spec.Disabled {
+		reqLogger.Info("Ensuring user and identity associated with UserAccount")
 		var createdOrUpdated bool
 		var user *userv1.User
 		if user, createdOrUpdated, err = r.ensureUser(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureIdentity(reqLogger, userAcc, user); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureNSTemplateSet(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-	} else if util.HasFinalizer(userAcc, userAccFinalizerName) {
-		if err = r.manageCleanUp(userAcc); err != nil {
+	} else if util.HasFinalizer(userAcc, userAccFinalizerName) || userAcc.Spec.Disabled {
+		reqLogger.Info("Deleting user and identity associated with UserAccount")
+		if err = r.deleteUserIdentity(userAcc); err != nil {
 			return reconcile.Result{}, err
 		}
+
+		if userAcc.Spec.Disabled {
+			disabled, err := r.checkAccountDisabled(userAcc)
+			if disabled {
+				return reconcile.Result{}, r.setStatusDisabled(userAcc)
+			} else {
+				return reconcile.Result{}, err
+			}
+		}
 	}
 	return reconcile.Result{}, r.setStatusReady(userAcc)
 }
 
+func (r *ReconcileUserAccount) checkAccountDisabled(userAcc *toolchainv1alpha1.UserAccount) (bool, error) {
```

Comment:
We don't really need `checkAccountDisabled`. Just return a bool in `deleteUserIdentity()`. It should return `false` if both User and Identity do not exist (so there is nothing to delete). Similar to what `deleteUser()`/`deleteIdentity()` functions do.

Reply:
done here: c0106baf040cff9a90957c721626bf27fb5e97bf